code/__HELPERS/unsorted.dm

```diff
@@ -1210,4 +1205,57 @@ will handle it, but:
 	if(location && include_turf) //At this point, only the turf is left, provided it exists.
 		. += location
 
+/**
+ *	Generates a cone shape. Any other checks should be handled with the resulting list. Can work with up to 359 degrees
+ *	Variables:
+ *	center - where the cone begins, or center of a circle drawn with this
+ *	max_row_count - how many rows are checked
+ *	starting_row - from how far should the turfs start getting included in the cone
+ *	cone_width - big the angle of the cone is
+ *	cone_direction - at what angle should the cone be made, relative to the game board's orientation
+ *	blocked - whether the cone should take into consideration solid walls
+ *	bypass_window - whether it will go through transparent windows like lasers
+ *	projectile - whether throwpass will be checked to ignore dense objects like projectiles
+ *	bypass_xeno - whether to bypass dense xeno structures like flamers
+ */
+/proc/generate_true_cone(atom/center, max_row_count = 10, starting_row = 1, cone_width = 60, cone_direction = 0, blocked = TRUE, bypass_window = FALSE, projectile = FALSE, bypass_xeno = FALSE)
+	var/right_angle = cone_direction + cone_width/2
+	var/left_angle = cone_direction - cone_width/2
+
+	//These are needed because degrees need to be from 0 to 359 for the checks to function
+	if(right_angle >= 360)
+		right_angle -= 360
+
+	if(left_angle < 0)
+		left_angle += 360
+
+	var/list/cardinals = GLOB.alldirs
+	var/list/turfs_to_check = list(get_turf(center))
+	var/list/cone_turfs = list()
+
+	for(var/r in 1 to max_row_count)
+		if(r > 2)
+			cardinals = GLOB.cardinals
+		for(var/X in turfs_to_check) //checks the inital turf, then afterwards checks every turf that is added to cone_turfs
+			var/turf/trf = X
+			for(var/direction in cardinals)
+				var/turf/T = get_step(trf, direction) //checks all turfs around X
```

Comment:
use AS, and please use proper var names this reads like algebra

Reply:
done